### ISSUE TITLE
add "addendum" and "howpublished" free-text annotation variables

### ISSUE THREAD

**Issue by denismaier** [citation-style-language/csl-evolution#16]
Biblatex has a couple of fields that can be used to add arbitrary pieces of information to an item: `howpublished` can be used to supply publishing information in a free format, `addendum` is used for the same but printed at the end of a reference. When I've used biblatex in the past these fields have  been very useful as there will always be cases that can't be cited properly otherwise.

Information given in that field can't be modified afterwards, of course. But I don't think that this is problem as using such mechanisms should be the last resort anyway.

I don't think there will be much impact on existing styles. We could either add those fields to the schema and not use them in the standard styles. Or, we could instruct processors to always print `addendum` at the end of each reference.

**Comment by bdarcus** [comment]
I agree.

I modified the title to be more assertive. Let me know if any objections?

**Comment by denismaier** [comment]
Good. Thanks.

**Comment by adam3smith** [comment]
I have significant concerns about this one and would like to veto this for 1.1 so that we can discuss it more thoroughly without holding things up. 

I think one of CSL's comparative strength is that it does a pretty good job to discourage style-specific data entry (which this will inevitably encourage). I also do see CSL as still principally oriented towards GUI products, where adding fields has real costs.
I'm not beyond being convinced, but I view this as a significant departure from previous CSL design ideas and I definitely don't think it should be in a release that's purposefully minor.

**Comment by denismaier** [comment]
I'm totally ok with postponing this if there's a need for more discussion–especially since it's already possible to use any variable with citeproc-js, and adding new variables to pandoc-citeproc is easy enough. (So feel free to remove the 1.1. label.)

Discouraging style-specific data entry is a good thing, for sure. But, on the other hand, there's the question if it is indeed possible to cover every possible use case. I think we should try to cover as many use cases as reasonably possible, but there will always be demands that cannot be satisfied, at least not at the moment. And therefore, I see three options:

1. Tell those users to use something else, type in their bibliographies manually, or to correct the results manually as a final step.
2. Tell users to use some hackish solutions, like using some otherwise not used item type or variable for that.
3. Implement a albeit hackish and imperfect solution ourselves.

(Are there more?)

Re 1. I don't think that is what we want to do. 
Re 2. This is what happens currently from time to time. (At least, that the impression I get from skimming the Zotero forum occasionally. I am not a as regular reader as you are, so please correct me if I am overstating the case here.) 

If this is correct (is it?), we should pursue something in the direction of 3. 
Again, doesn't need to be right now, and there might be better solutions than introducing these two variables.

**Comment by bdarcus** [comment]
I think you both raise good points. My impulse is to add it. But the decision might be easier if we had a couple of examples of why people need it?

**Comment by bwiernik** [comment]
In standard citation styles, my understanding is that `howpublished` is mostly analogous to `genre` or `medium`. It is only used for `@misc` and typically contains information about the kind of item (e.g., "Webpage"; cf. https://tex.stackexchange.com/questions/186235/in-bibtex-when-should-i-use-howpublished-and-when-url). 

I don't really see how `addendum` adds anything beyond `note` and `annote`. `note` and `annote` can already fill the "arbitrary information added to a reference" role well (e.g., `note` is used in annotated bibliography styles; two places I've seen `annote` used is (1) to add comments about author arrangements, awards, etc. on CVs; (2) to note which bibliography entries contributed data to a review article or meta-analysis). 

One place I've seen `addendum` used is to provide information about reprinted versions. apa.csl uses `references` for that. 

In short, I really don't see the value in adding them, and I see significant downsides as Sebastian notes.

**Comment by denismaier** [comment]
Good points. `genre` and `medium` can be used to some degree.
Concerning `note` and `annote`.
`annote` could work, but it is defined as "reader’s notes about the item content". (I don't use it that way so abusing it would be possible...)
`note`: that's just the extra field in Zotero right? Can I use this field for `original publisher` stuff and still add free-text information?

**Comment by bwiernik** [comment]
About `annote`, it isn't used in any repository style, and no client has a field mapped to it. The only really widespread application I know is the apa-meta-analysis style that I distribute and that is used by the papaja R package to annotate meta-analysis bibliographies. That choice was made given that `note` (which has a more conceptually-aligned definition) is used by Zotero `extra`, Mendeley `note`, and other fields. 

I think we ought to update the spec to define `annote` as annotations on the bibliography entry. `note` is for things like user notes about the item. As currently defined, `note` and `annote` are redundant, and clients already use `note` in a way that more aligns with the current `annote` definition. 

In Zotero, you can add CSL fields to Extra like this: `original-publisher: Publisher` and still have it contain free-text information. This is a feature of citeproc-js, which pulls out CSL variables from `note` and adds them to the item data (before rendering `note` if needed).

Beyond that, we should not expect any normal style to use `note`. It often contains a variety of pieces of information, and it should not be expected to be formatted in a style-specific manner except in extraordinary circumstances (e.g., an annotated bibliography style can expect that `note` contains the user's notes about the item). If a user needs arbitrary text entered in `note`, we should expect them to edit the CSL style.

**Comment by denismaier** [comment]
Ok, so `annote` is actually what I wanted `addendum` to be.  And for the cases where `medium` is not enough I can still use `note` as a last resort.
That's good for me and this can be closed.

**Comment by bdarcus** [comment]
Could you change the title to reflect the current state?

**Comment by denismaier** [comment]
I have already opened a new issue (https://github.com/citation-style-language/csl-evolution/issues/28) with an appropriate title. Should I close that one and keep this one here; or keep the other and close this?
